### PR TITLE
Add filename parameter to data queries

### PR DESCRIPTION
**Pull request by efronbs**
Issue: https://github.com/yahoo/fili/issues/709

This PR only exposes the filename parameter to data queries, though `getDownloadFilename()` was added to the `ApiRequest` interface. If you think more query types should have this parameter exposed please comment.

### CODE REVIEW

**Comment by efronbs** [comment]
I want to manually test the new parameter before merging

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java [hunk 1]
```diff
@@ -87,6 +87,7 @@
     protected final ResponseFormatType format;
     protected final Optional<PaginationParameters> paginationParameters;
     protected final long asyncAfter;
+    protected final Optional<String> downloadFilename;
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java]: Don't store as Optional.  Instead store as potentially null and unpack to optional on the way out.

Comment (efronbs) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java]: why not store as optional?

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/HttpResponseMaker.java [hunk 2]
```diff
@@ -267,16 +265,35 @@ protected ResponseBuilder buildAndAddResponseHeaders(
...
+
+    /**
+     * Builds the headers for the response and gives them to the provided response builder to be added to the response.
+     *
+     * @param rspBuilder  ResponseBuilder that handles adding the headers to the response
+     * @param responseFormatType  The type of the response
+     * @param containerRequestContext  The request context
+     * @param downloadFilename  The filename the response should be downloaded as. Null or empty indicates the response
+     * should not be downloaded and instead rendered by the browser
+     * @return the response builder that has had the headers added
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/HttpResponseMaker.java]: vertical whitespace between @params and @return

**Pull request merged**